subsys/bluetooth/crypto/CMakeLists.txt

```diff
@@ -4,6 +4,10 @@ zephyr_library()
 
 zephyr_library_sources(bt_crypto.c)
 
+if(CONFIG_BT_USE_PSA_API)
+    zephyr_link_libraries(mbedTLS)
```

Comment:
Why is this required if `CONFIG_BT_USE_PSA_API` is enabled for platform with support of TF-M based PSA?

Reply:
This is only because I need `psa/crypto.h` header to be available. I tried to use the TF-M one in another PR, but without much success. I can give it another try though

Reply:
I tried replacing `zephyr_link_libraries(mbedTLS)` with
```
zephyr_include_directories(
        $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>/api_ns/interface/include
    )
```
but the build fails due to the fact that:
```
zephyr/build/tfm/api_ns/interface/include/mbedtls/build_info.h:106:10: fatal error: mbedtls/mbedtls_config.h: No such file or directory
  106 | #include "mbedtls/mbedtls_config.h"
```

Side note: the same happens in `samples/bluetooth/mesh/` if I set `CONFIG_MBEDTLS=n` in `prj.conf`.

Reply:
Shouldn't they be added together as it has been done in mesh:
```
zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)

if (CONFIG_BUILD_WITH_TFM)
  target_include_directories(${ZEPHYR_CURRENT_LIBRARY} PRIVATE
    $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>/api_ns/interface/include
  )
endif()
```
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/CMakeLists.txt#L128-L134

When I, as user, choose TF-M I do not configure anything related to mbedTLS (and even do not choose it explicitly) and vice versa.